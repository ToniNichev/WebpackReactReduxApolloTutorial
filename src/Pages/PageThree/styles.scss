.wrapper {
  .canvas {
    border: 1px solid blue;
  }

  .symbolTitle {
    color:blue;
  }

  .QuoteStrip {
    background-color: black;
    border: 1px solid red;
    display: flex;
  }
}