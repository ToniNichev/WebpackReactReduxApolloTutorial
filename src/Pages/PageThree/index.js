import React, { useState, useEffect } from 'react';
import DownloadChartButton from '../../components/ChartModules/DownloadChartButton';
import DownloadChartPopup from '../../components/ChartModules/DownloadChartPopup';
import Chart from '../../images/chart.png';
import styles from './styles.scss';

const quoteData =  {
  //curmktstatus: 'POST_MKT',
  curmktstatus: 'REG_MKT',
  ExtendedMktQuote: {
    last: '115.80',
    last_timedate: "7:59 PM EDT",
    change: '+10.5',
    change_pct: '+12.4'
  },
  last: '110.78',
  change: "+1.34",
  change_pct: "+1.23",
  name: "Apple Inc",
  symbol: "AAPL",
  exchange: "NASDAQ",
  currencyCode: "USD",
  last_timedate: "4:00 PM EDT",
}

const chartContainer = 'chartContainer';
const chartTimeIntervalSelected = 'chartTimeIntervalSelected';

function PageThree() {
  
  useEffect(() => {
    const chartCanvas = document.getElementById('chartCanvas');
    if(chartCanvas === null)
      return;
    const canvasContext = chartCanvas.getContext('2d');
    var image = new Image();
    image.src = Chart;
    image.onload = function() {
      canvasContext.drawImage(image, 0, 0);
    };
  });

  const [downloadImagePopupVisible, setValueOfDownloadImagePopupVisible] = useState(false);

  return(
    <div className={styles.wrapper}>
      <div className={chartTimeIntervalSelected}>1D</div>
      <div className={styles.QuoteStrip}>
        <div className={styles.symbolTitle}>Apple Inc</div>
        <DownloadChartButton sendData={() => { setValueOfDownloadImagePopupVisible(!downloadImagePopupVisible); }} />
      </div>
      <div className={chartContainer}>
        <canvas height="350" width="678" id="chartCanvas" className={styles.canvas}/>
      </div>        
      { downloadImagePopupVisible && <DownloadChartPopup 
                                      quoteData={quoteData} 
                                      sendData={() => { setValueOfDownloadImagePopupVisible(!downloadImagePopupVisible); }} /> }
    </div>);
}
export default PageThree;