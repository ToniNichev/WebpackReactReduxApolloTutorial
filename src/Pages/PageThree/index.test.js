import React from 'react';
import { shallow, mount } from 'enzyme';
import PageThree from './';
import toJson from 'enzyme-to-json';

const quoteData =  {
  ExtendedMktQuote: {
    last_timedate: "7:59 PM EDT",
  },
  name: "Apple Inc",
  symbol: "AAPL",
  exchange: "NASDAQ",
  currencyCode: "USD",
  last_timedate: "4:00 PM EDT",
}

let chartCanvas;

global.isJestEnv = true;

describe('Testing PageThree component', () => {

    it('renders as the snapshot', () => {
      const wrapper = mount(
        <PageThree />
      );
     expect(toJson(wrapper)).toMatchSnapshot();
   });  



   it('button click adds window.location.hash', () => {
     const wrapper = mount(
      <PageThree />
     );

     
     //wrapper.find('button').simulate('click');
     console.log("!!>>>", wrapper.find('button'));
     //console.log(">>>>", wrapper.find('.DownloadChartPopup-downloadChartModal').html());     
     //wrapper.find('button').simulate('click');
     //console.log(">>>>", wrapper.find('.DownloadChartPopup-downloadChartModal').html());
     
     //expect(wrapper.find('.DownloadChartPopup-downloadChartModal')).toBe('one');
   });   
});
