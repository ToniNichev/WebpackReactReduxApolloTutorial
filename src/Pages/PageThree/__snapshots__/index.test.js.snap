// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component renders as the snapshot 1`] = `
<PageThree>
  <div>
    <div
      className="chartTimeIntervalSelected"
    >
      1D
    </div>
    <div>
      <div>
        Apple Inc
      </div>
      <DownloadChartButton
        sendData={[Function]}
      >
        <div>
          <button
            onClick={[Function]}
          >
            <img
              alt="download chart"
              src={Object {}}
            />
            EXPORT
          </button>
        </div>
      </DownloadChartButton>
    </div>
    <div
      className="chartContainer"
    >
      <canvas
        height="350"
        id="chartCanvas"
        width="678"
      />
    </div>
  </div>
</PageThree>
`;

exports[`Testing PageThree component renders as the snapshot 1`] = `
<PageThree>
  <div>
    <div
      className="chartTimeIntervalSelected"
    >
      1D
    </div>
    <div>
      <div>
        Apple Inc
      </div>
      <DownloadChartButton
        sendData={[Function]}
      >
        <div>
          <button
            onClick={[Function]}
          >
            <img
              alt="download chart"
              src={Object {}}
            />
            EXPORT
          </button>
        </div>
      </DownloadChartButton>
    </div>
    <div
      className="chartContainer"
    >
      <canvas
        height="350"
        id="chartCanvas"
        width="678"
      />
    </div>
  </div>
</PageThree>
`;
